File: exercism/typescript/allergies/allergies.ts
export class Allergies {
  static allergens: string[] = ["eggs", "peanuts", "shellfish", "strawberries", "tomatoes", "chocolate", "pollen", "cats"];
  allAllergies: string[];

  constructor(allergenIndex: number) {
    let binaryArray: number[] = allergenIndex.toString(2).split("").map(x => parseInt(x)).reverse();
    this.allAllergies = [];
    for (let i = 0; i < binaryArray.length; i++) {
      if (i <= 7 && binaryArray[i] === 1) {
        this.allAllergies.push(Allergies.allergens[i]);
      }
    }
  }

  public list(): string[] {
    return this.allAllergies;
  }

  public allergicTo(allergen: string): boolean {
    return this.allAllergies.find(element => element === allergen) !== undefined;
  }
}

File: exercism/python/allergies/allergies.py
"""Allergies exercise implementation."""


class Allergies:
    """Represent allergies encoded in an integer score."""

    _ALLERGENS = (
        "eggs",
        "peanuts",
        "shellfish",
        "strawberries",
        "tomatoes",
        "chocolate",
        "pollen",
        "cats",
    )
    _VALID_BITS_MASK = 0xFF

    def __init__(self, score):
        """Store only score bits that map to known allergens."""
        self._score = score & self._VALID_BITS_MASK

    def allergic_to(self, item):
        """Return True when the encoded score contains the given allergen."""
        if item not in self._ALLERGENS:
            return False

        allergen_index = self._ALLERGENS.index(item)
        allergen_value = 1 << allergen_index
        return bool(self._score & allergen_value)

    @property
    def lst(self):
        """Return all allergens present in the encoded score."""
        return [
            allergen
            for allergen_index, allergen in enumerate(self._ALLERGENS)
            if self._score & (1 << allergen_index)
        ]
